### PR TITLE
feat: add redis compat client

### PR DESCRIPTION
**Pull request by pgautier404**

### CODE REVIEW

File: README.template.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+don't support yet, we provide our own `MomentoRedisBase` abstract base class which explicitly lists out 
+the APIs we currently support. Simply use the class as a type annotation for your client:
+
+```python
+from momento_python_redis_client import MomentoRedis, MomentoRedisBase
+redis_client: MomentoRedisBase = MomentoRedis(...)
+```
+
+Once the client is typed using the abstract base class, static analysis tools will allow you to find 
+calls to as yet unsupported APIs.
```
Comment (cprice404) [inline README.template.md:120]: yay

File: .github/workflows/on-push-to-main-branch.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+      - name: Publish
+        env:
+          POETRY_PYPI_TOKEN_PYPI: ${{ secrets.PYTHON_CUSTOMER_SDK_PYPI_TOKEN }}
+        run: |
+          echo "poetry version ${{ steps.release.outputs.major }}.${{ steps.release.outputs.minor }}.${{ steps.release.outputs.patch }}"
+#          curl -sL https://install.python-poetry.org | python - -y --version 1.3.1
+#          poetry version ${{ steps.release.outputs.major }}.${{ steps.release.outputs.minor }}.${{ steps.release.outputs.patch }}
+#          poetry build
+#          poetry publish
```
Comment (cprice404) [inline .github/workflows/on-push-to-main-branch.yml:102]: nice

**Review by cprice404 — APPROVED** [review]
whoohoo

File: tests/test_clients.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+skip_redis = os.getenv("TEST_SKIP_REDIS", False)
+skip_momento = os.getenv("TEST_SKIP_MOMENTO", False)
+
+
+def _use_client(client: str) -> bool:
+    if client == "momento_redis_client" and skip_momento:
+        return False
+    elif client == "redis_client" and skip_redis:
+        return False
+    return True
```
Comment (malandis) [inline tests/test_clients.py:24]: Took me a minute to understand the default was "run both" we can independently opt out of each one. I think the code below may be more maintainable if we have one environment variable to toggle between the momento v redis; the tests only use one client (can get rid of the if tests and parametrized thing); then run the tests twice.

Not a blocker but something to consider for a future PR

**Review by malandis — APPROVED** [review]

**Review by malandis — COMMENTED** [review]
^ Only question is about library vs dev dependencies.

File: tests/test_clients.py (hunk 3 above)
Comment (pgautier404) [reply tests/test_clients.py:24]: We can consider changing these up, but I really like having the ability to run them both at the same time so we can easily and simultaneously verify both the internal consistency of each library as well as the consistency of results between them without having to collate results from multiple runs. I'm going to leave them as they are for now but am totally open to having a conversation about improving this in the future.

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+readme = "README.md"
+license = "Apache-2.0"
+
+[tool.poetry.dependencies]
+python = "^3.7,<3.12"
+momento = "1.7.1"
+redis = "4.6.0"
+typing-extensions = "4.5.0"
+types-redis = "^4.6.0.4"
+types-protobuf = "^4.24.0.1"
```
[the remark this answers is not in the document]
Comment (pgautier404) [reply pyproject.toml]: Good call, thanks. I moved the redis and protobuf type stubs to dev dependencies. The `typing-extensions` dependency is actually used in code so was left where it is.

**Review by malandis — APPROVED** [review]

**Pull request merged**